dexlib2/src/main/java/com/android/tools/smali/dexlib2/builder/MutableMethodImplementation.java

```diff
@@ -256,17 +252,21 @@ public List<BuilderInstruction> getInstructions() {
         if (fixInstructions) {
             fixInstructions();
         }
-        return Iterables.concat(
-                Iterables.transform(instructionList, new Function<MethodLocation, Iterable<? extends DebugItem>>() {
-                    @Nullable @Override public Iterable<? extends DebugItem> apply(@Nullable MethodLocation input) {
-                        assert input != null;
-                        if (fixInstructions) {
-                            throw new IllegalStateException("This iterator was invalidated by a change to" +
-                                    " this MutableMethodImplementation.");
-                        }
-                        return input.getDebugItems();
-                    }
-                }));
+
+        ArrayList<DebugItem> debugItems = new ArrayList<>();
+
+        for (MethodLocation methodLocation: instructionList) {
+            if (methodLocation == null) {
+                throw new NullPointerException();
```

Comment:
Just keep this as an assert? It will NPE below anyway.

Reply:
Done